File: packages/hmssdk_flutter/ios/Classes/Models/HMSHLSVariantExtension.swift
//
//  HMSHLSVariantExtension.swift
//  hmssdk_flutter
//
//  Created by Govind Maheshwari on 19/01/22.
//

import Foundation
import HMSSDK

class HMSHLSVariantExtension {

    static func toDictionary(_ hmshlsVariant: HMSHLSVariant) -> [String: Any] {

        var dict = [String: Any]()

        dict["meeting_url"] = hmshlsVariant.meetingURL.absoluteString

        dict["metadata"] = hmshlsVariant.metadata

        if let url = hmshlsVariant.url {
            dict["hls_stream_url"] = url.absoluteString
        } else {
            dict["hls_stream_url"]  = nil
        }

        if let startedAt = hmshlsVariant.startedAt {
            dict["started_at"] =  Int(startedAt.timeIntervalSince1970 * 1000)
        }

        if let playlistType = hmshlsVariant.playlistType {
            dict["playlist_type"] = getStringFromHMSHLSPlaylistType(playlistType: playlistType)
        }

        return dict
    }

    private static func getStringFromHMSHLSPlaylistType(playlistType: HMSHLSPlaylistType) -> String {
        switch playlistType {
            case .dvr:
                return "dvr"
            case .noDVR:
                return "noDvr"
            default:
                return "noDvr"
        }
    }

}

File: packages/hmssdk_flutter/android/src/main/kotlin/live/hms/hmssdk_flutter/HMSHLSVariantExtension.kt
package live.hms.hmssdk_flutter

import live.hms.video.sdk.models.HMSHLSPlaylistType
import live.hms.video.sdk.models.HMSHLSVariant

class HMSHLSVariantExtension {
    companion object {
        fun toDictionary(hmshlsVariant: HMSHLSVariant?): HashMap<String, Any>? {
            val args = HashMap<String, Any>()
            if (hmshlsVariant == null)return null
            args["hls_stream_url"] = hmshlsVariant.hlsStreamUrl ?: ""
            args["meeting_url"] = hmshlsVariant.meetingUrl ?: ""
            args["metadata"] = hmshlsVariant.metadata ?: ""
            hmshlsVariant.startedAt?.let {
                args["started_at"] = it
            }
            args["playlist_type"] = getHMSHLSVariantToString(hmshlsVariant.playlistType)
            return args
        }

        private fun getHMSHLSVariantToString(hmsHlsVariantType: HMSHLSPlaylistType?): String {
            return when (hmsHlsVariantType) {
                HMSHLSPlaylistType.dvr -> {
                    "dvr"
                }
                HMSHLSPlaylistType.noDVR -> {
                    "noDvr"
                }
                else -> {
                    "noDvr"
                }
            }
        }
    }
}
